src/scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -472,7 +469,33 @@ def _highly_variable_genes_batched(
             missing_hvg["gene"] = gene_list[~filt]
             hvg = pd.concat([hvg, missing_hvg], ignore_index=True)
 
-        dfs.append(hvg)
+        return hvg
+
+    sanitize_anndata(adata)
+    batches = adata.obs[batch_key].cat.categories
+    X = _get_obs_rep(adata, layer=layer)
+
+    per_batch_func = process_batch
+    if isinstance(X, DaskArray):
+        from dask import compute, delayed
+
+        per_batch_func = delayed(process_batch)
+        _set_obs_rep(adata, X.persist(), layer=layer)
+
+    dfs = [
+        per_batch_func(
+            batch_mask=adata.obs[batch_key] == batch,
+            adata=adata,
+            layer=layer,
+            gene_list=adata.var_names,
+            cutoff=cutoff,
+            n_bins=n_bins,
+            flavor=flavor,
+        )
+        for batch in batches
+    ]
+    if not isinstance(dfs[0], pd.DataFrame):
+        dfs = compute(*dfs)
```

Comment:
I think this is cleaner:

Do

```py
if is_dask := isinstance(X, DaskArray):
    from dask import delayed
```

above and then

```suggestion
    if is_dask:
        from dask import compute

        dfs = compute(*dfs)
```

maybe even do `dfs = (compute(df) for df in dfs)`?

if there are many, pandas might be able to use that more effectively, but maybe have dask schedule all the computes as it sees fit is better.